en/render-pipeline/custom-pipeline.md

```diff
@@ -0,0 +1,243 @@
+
+Each type will have its own different **RenderPass**.
+
+### Raster
+
+The raster type uses the rasterization capabilities of the GPU (executed in GraphicsEngine).
+
+#### 1. RasterPass
+
+<img src="./image/cp-raster-pass.png" width=760></img>
+
+- width, height is the resolution of the output rendering target.
+
+- layoutName is the name of the Effect's Stage.
+
+- name is the name displayed during debugging. If it is empty, the system will give the default name.
+
+#### 2. RasterSubpass
+
+Feature not yet open. Requires GPU Tile-based rendering capability.
+
+#### 3. PresentPass
+
+Render the screen to the screen.
+
+### Compute
+
+The calculation type uses the general-purpose computing power of the GPU, as well as ray-tracing capabilities (executable in GraphicsEngine, ComputeEngine).
+
+#### 1. ComputePass
+
+<img src="./image/cp-compute-pass.png" width=760></img>
+
+- layoutName is the name of the Effect's Stage.
+
+- name is the name displayed during debugging. If it is empty, the system will give the default name.
+
+#### 2. RaytracePass
+
+Feature not yet open. Requires GPU ray tracing capability.
+
+### Resource
+
+The resource type uses the resource processing power of the GPU (executable in GraphicsEngine, ComputeEngine, CopyEngine).
+
+#### 1. CopyPass
+
+Responsible for copying the resource source to the target, which requires the resource format to be compatible.
+
+<img src="./image/cp-copy-pass.png" width=760></img>
+
+- name is the name displayed during debugging. If it is empty, the system will give the default name.
+
+#### 2. MovePass
+
+Responsible for moving the resource source to the target, requiring the resource to be in the same format.
+
+Move here is a semantic concept (move semantics): move the source variable to the target variable, nullifying the source variable. If the resource cannot be moved for some reason (e.g. the resource source is being read), this is done as a copy.
+
+Move semantics are used for pipeline optimization for the purpose of bandwidth reduction. If you are not sure how to use **Move Pass** properly, you can use **Copy Pass** instead, which will not affect the screen performance and is easier to debug.
+
+## RenderView
+
+There are two types of RenderView: **RasterView**, **ComputeView**.
+
+### RasterView
+
+**Rasterized view** will be rasterized. There are two subtypes: RenderTarget, DepthStencil.
+
+<img src="./image/cp-raster-view.png" width=560></img>
+
+- slotName is the name of the shader pixel component. (e.g. color, normal, etc.)
+
+- accessType is the binding type, which can be Read, ReadWrite, Write, Read as input, Write as output, ReadWrite as both input and output, [Note] DepthStencil When doing DepthTest, although the result is not written to the view, but as output, the binding type is still Write. some platforms open ARM_shader_framebuffer_fetch_depth_stencil extension, DepthStencil binding type is ReadWrite. The DepthStencil binding type cannot be Read.
```

Comment:
some platforms open ARM_shader_framebuffer_fetch_depth_stencil extension, DepthStencil binding type is ReadWrite.-> When ARM_shader_framebuffer_fetch_depth_stencil extension is enabled in a shader, the corresponding DepthStencil binding type should be set as ReadWrite.